### PR TITLE
upload new colours for barcode/spine legend jpg

### PR DESCRIPTION
**Pull request by vaelliott**
New colours for the spine/barcode plot vertical lines

### CODE REVIEW

File: server.R [hunk 1]
```diff
@@ -741,206 +741,200 @@ showModal(welcome_modal)
...
+      spine <- merge(spine_allareas(), spine_chosencomp(), by=c("indicator"))
+      spine <- left_join(spine, spine_chosenarea(), by=c("indicator"))
       
       #add variable denoting if sign diff between comparator
-      bar<-bar %>%
+      spine<-spine %>%
         mutate(flag=case_when(
-          bar$interpret == "O" ~ 'No significance can be calculated',
-          bar$lowci_chosen<=bar$measure_comp & bar$upci_chosen>=bar$measure_comp ~
+          spine$interpret == "O" ~ 'No significance can be calculated',
```
Comment (jvillacampa) [inline server.R]: I  don't think you need to specify the dataset all the time, e.g. spine$interpret could be interpret.

File: server.R [hunk 2]
```diff
@@ -741,206 +741,200 @@ showModal(welcome_modal)
...
-      #Transposing data so that better is always to the right of plot
-      bar <- bar %>%
+      #Transposing data so that better is always to the right hand side of plot
+      spine <- spine %>%
         mutate(comp = 1,
-               all = bar$measure/bar$measure_comp,
-               chosen = bar$measure_chosen/bar$measure_comp,
-               all2=case_when(bar$interpret=='L' & bar$measure>bar$measure_comp ~ -(all-1),
-                              bar$interpret=='L' & bar$measure<=bar$measure_comp ~ 
+               all = spine$measure/spine$measure_comp,
```
Comment (jvillacampa) [inline server.R]: same as above

File: global.R [hunk 3]
```diff
@@ -75,6 +75,18 @@ plot_nodata <- function() {
     config( displayModeBar = FALSE) # taking out plotly logo and collaborate button
 }
 
+#Function to create plot when no data available for ggplot visuals
+plot_nodata_gg <- function() {
+  ggplot()+
+    xlab("No data available")+
+    scale_x_discrete(position = "top")+
+    theme(panel.background = element_blank(),
+          axis.title.x=element_text(size=12, colour ='#555555'))
```
Comment (jvillacampa) [inline global.R]: very small font size, move to 20

File: server.R [hunk 4]
```diff
@@ -952,103 +946,111 @@ showModal(welcome_modal)
...
-      bar<-bar %>%
+      spine<-spine %>%
         mutate(flag=case_when(
-          bar$interpret == "O" ~'NA',
-          bar$lowci_chosen<=bar$measure_comp & bar$upci_chosen>=bar$measure_comp ~ 'NS',
-          bar$lowci_chosen > bar$measure_comp & bar$interpret == "H" ~ 'Better',
-          bar$lowci_chosen > bar$measure_comp & bar$interpret == "L" ~ 'Worse',
-          bar$upci_chosen < bar$measure_comp & bar$interpret == "L" ~ 'Better',
-          bar$upci_chosen < bar$measure_comp & bar$interpret == "H" ~ 'Worse',
+          spine$interpret == "O" ~'NA',
```
Comment (jvillacampa) [inline server.R]: again no need of spine$

File: ui.R [hunk 5]
```diff
@@ -327,38 +327,56 @@ tabPanel("Heatmap", icon = icon("list-ul"), value = "heat",
...
+                    "Worse than comparator", br(),
+                    img(src='signif_nocalc.png', height=18, style="padding-right: 2px; vertical-align:middle"), 
+                    "No differences can be calculated")),
+           column(5,
+                  br(),
+                  br(),
+                  uiOutput("ui_spine_legend_selected"),
+                  uiOutput("ui_spine_legend_areatype"),
+                  uiOutput("ui_spine_legend_comparator"))),
+         mainPanel(
```
Comment (jvillacampa) [inline ui.R]: add width = 12 so it covers the whole space. and if you want margins change the dith % in the plotOutput

File: ui.R [hunk 6]
```diff
@@ -327,38 +327,56 @@ tabPanel("Heatmap", icon = icon("list-ul"), value = "heat",
...
+           withSpinner(uiOutput("ui_spine_plot"))
```
Comment (jvillacampa) [inline ui.R]: to ensure it's centered you can do:
           div(align = "center", withSpinner(uiOutput("ui_spine_plot")))

**Comment by jvillacampa** [comment]
I have made some comments to specific parts of the code. It works fine, and I think it's almost ready. I would move "select you area" under geography level, and have the wellPanel arranged in 4 columns, or potentially 3.

File: ui.R (hunk 5 above)
Comment (vaelliott) [reply ui.R]: done

File: ui.R (hunk 6 above)
Comment (vaelliott) [reply ui.R]: I added this line of code but it didn't seem to make any difference to the alignment. It looks like the chart is left aligned which seems ok to me so I was just going to ignore this for now.

File: server.R (hunk 4 above)
Comment (vaelliott) [reply server.R]: ok

File: server.R (hunk 2 above)
Comment (vaelliott) [reply server.R]: done

File: server.R (hunk 1 above)
Comment (vaelliott) [reply server.R]: ok

**Comment by jvillacampa** [comment]
Good to go

**Pull request merged**